src/services/dynamo-db-service.ts

```diff
@@ -103,4 +89,26 @@ export class DynamoDbService {
     logger.debug(JSON.stringify(response));
     return response;
   }
+
+  public async updateDeleteStatus(userId: string) {
+    const ttl = getCurrentTimestamp().seconds + appConfig.maxRetentionSeconds;
+    const commandInput: UpdateItemCommandInput = {
+      TableName: this.tableName,
+      Key: { pk: { S: userId } },
+      UpdateExpression: 'SET #isAccountDeleted = :isAccountDeleted, #ttl = :ttl',
+      ExpressionAttributeNames: {
+        '#isAccountDeleted': 'isAccountDeleted',
+        '#ttl': 'ttl',
+      },
+      ExpressionAttributeValues: {
+        ':isAccountDeleted': { BOOL: true },
+        ':ttl': { N: ttl.toString() },
+      },
+      ConditionExpression: 'attribute_exists(pk)',
+    };
+    const command = new UpdateItemCommand(commandInput);
+    const response = await this.dynamoClient.send(command);
+    logger.debug(JSON.stringify(response));
```

Comment:
if we are wanting to keep this line here, can we please use the sensitive info prefix? As it may contain user id I think